src/Features/Core/Portable/MetadataAsSource/MetadataAsSourceFileService.cs

```diff
@@ -159,17 +172,21 @@ public bool ShouldCollapseOnOpen(string? filePath, BlockStructureOptions blockSt
             if (filePath is null)
                 return false;
 
-            var workspace = GetWorkspaceOnMainThread();
+            var workspace = _workspace;
+            Debug.Assert(workspace != null, "This should only be called by outlining once outlining has already confirmed that this is a file from the MetadataAsSourceWorkspace");
 
-            foreach (var provider in _providers)
+            if (workspace != null)
```

Comment:
if the conditional is needed, then maybe consider logging a NFW here

Reply:
Idk if we'd want an NFW unless this is actually an error state. The comments make it seem like we expect this to be a reasonable request and if the workspace isn't created yet then no work is needed

Reply:
sure :)

Reply:
done.

Reply:
The message in assert makes it sound like this could be called before MetadataAsSource workspace is created if the caller somehow messed it up, which would indicate an error state.

Reply:
this one case is an error we'd want to fix.  The other two are totally reasonable as part of our contract  here.  So i'm only NFWing in teh one location (the same one that was originally an assert).